This project uses the Kaggle Dataset ["When do children learn words?"](https://www.kaggle.com/rtatman/when-do-children-learn-words)

In [1]:
import pandas as pd #data manipulation & analysis 

# Data Initialization & Exploration

In [3]:
#import the "main_data" CSV from Data folder
main_data_df = pd.read_csv("Data/main_data.csv") 

#import the "Norwegian_CDS_frequency" CSV from Data folder 
Norwegian_CDS_frequency_df = pd.read_csv("Data/Norwegian_CDS_frequency.csv")

## main_data_df feature & metadata EDA 

In [4]:
main_data_df.head()

ID_CDI_I ID_CDI_II           Word_NW          Word_CDI    Translation   AoA  \
0    i_4_1     i_1_1              'au'              'au'         'ouch'  16.0   
1    i_4_2     i_1_2             'bææ'             'bææ'      'baa baa'  15.0   
2    i_4_3     i_1_3  'brrr (bil-lyd)'  'brrr (bil-lyd)'        'vroom'  13.0   
3    i_4_4     i_1_4       'gakk gakk'       'gakk gakk'  'quack quack'  17.0   
4    i_4_5     i_1_5             'grr'             'grr'          'grr'  22.0   

    VSoA        Lex_cat Broad_lex    Freq  CDS_freq  
0   40.0  sound effects  nominals  4366.0       7.0  
1   40.0  sound effects  nominals    18.0       5.0  
2   20.0  sound effects  nominals     NaN      20.0  
3   40.0  sound effects  nominals    16.0       3.0  
4  220.0  sound effects  nominals    78.0       1.0

In [7]:
main_data_df.dtypes

ID_CDI_I        object
ID_CDI_II       object
Word_NW         object
Word_CDI        object
Translation     object
AoA            float64
VSoA           float64
Lex_cat         object
Broad_lex       object
Freq           float64
CDS_freq       float64
dtype: object

In [9]:
main_data_df.shape

(731, 11)

In [13]:
main_data_df.columns

Index(['ID_CDI_I', 'ID_CDI_II', 'Word_NW', 'Word_CDI', 'Translation', 'AoA',
       'VSoA', 'Lex_cat', 'Broad_lex', 'Freq', 'CDS_freq'],
      dtype='object')

In [12]:
main_data_df.isna().sum()

ID_CDI_I       341
ID_CDI_II        0
Word_NW          0
Word_CDI         0
Translation      0
AoA             36
VSoA            27
Lex_cat         16
Broad_lex       16
Freq            10
CDS_freq         8
dtype: int64

This dataset contains 731 enteries, 11 columns. There are missing enteries in 7 columns, that will need to be handled before analysis. Reference for columns: 
* IDCDII: Word ID from the Norwegian adaptation of the MacArthur-Bates Communicative Development Inventories, version 1
* IDCDIII: Word ID from the Norwegian adaptation of the MacArthur-Bates Communicative Development Inventories, version 2
* Word_NW: The word in Norwegian
* Word_CDI: The form of the word found in the Norwegian adaptation of the MacArthur-Bates Communicative Development Inventories
* Translation: the English translation of the Norwegian word
* AoA: how old a child generally is was when they this this word, in months (Estimated from the MacArthur-Bates Communicative Development Inventories)
* VSoA: how many other words a child generally knows when they learn this word (rounded up to the nearest 10)
* Lex_cat: the specific part of speech of the word
* Broad_lex: the broad part of speech of the word
* Freq: a measure of how commonly this word occurs in Norwegian
* CDS_Freq: a measure of how commonly this word occurs when a Norwegian adult is talking to a Norwegian child

## Norwegian_CDS_frequency_df feature & metadata EDA

In [15]:
Norwegian_CDS_frequency_df.head()

Word_CDI    Translation  Freq_NoWaC  Freq_CDS
0              'au'         'ouch'      4366.0       7.0
1             'bææ'      'baa baa'        18.0       5.0
2  'brrr (bil-lyd)'        'vroom'         NaN      20.0
3       'gakk gakk'  'quack quack'        16.0       3.0
4             'grr'          'grr'        78.0       1.0

In [18]:
Norwegian_CDS_frequency_df.dtypes

Word_CDI        object
Translation     object
Freq_NoWaC     float64
Freq_CDS       float64
dtype: object

In [19]:
Norwegian_CDS_frequency_df.shape

(731, 4)

In [21]:
Norwegian_CDS_frequency_df.columns

Index(['Word_CDI', 'Translation', 'Freq_NoWaC', 'Freq_CDS'], dtype='object')

In [23]:
Norwegian_CDS_frequency_df.isna().sum()

Word_CDI        0
Translation     0
Freq_NoWaC     10
Freq_CDS        8
dtype: int64

This dataset contains 731 entries and 4 columns. 2 columns have null values that will have to be resolved before analysis. Reference for columns: 
* Word_CDI: The word from, as found in the Norwegian adaptation of the MacArthur-Bates Communicative Development Inventories
* Translation: The English translation of the Norwegian word
* Freq_NoWaC: How often this word is used on the internet
* Freq_CDS: How often this word is used when talking to children (based on two Norwegian CHILDES corpora)